### DISCUSSION TITLE
How do you scan your documents?

### DISCUSSION THREAD

**Discussion by astubenbord** [astubenbord/paperless-mobile#118] [General]
...or to be more specific: What features would you like to see in a document scanner in this app?

As this is on the roadmap for quite some time now and it has been requested from some of you, I'd like to implement a document scanner which is tailored to scanning A4 documents for paperless. Now, since there are quite a lot of good document scanners out there (like Office Lens), I think it wouldn't make a lot of sense to implement yet another documents scanner (and honestly, I wouldn't event get close to the quality most of the popular options offer). Therefore I'm looking for some input on what you think would make a good document scanner specifically for usage with paperless.

I'm looking forward to your suggestions!

Cheers

**Comment by SpicyWasab** [comment]
Hi !
I have some ideas :

- OCR : Sometimes on some installations OCR is too heavy to do and the docs recommands "doing OCR on the client".
- Having an hint on the document (like a blue rectangle or something like that)
- - I think it's already the case since I saw it appearing sometimes, but for me it was kinda buggy and never placed onto the document

Here's some other ideas that would be useful for bulk scanning documents (like what I'm doing rn) :
- Auto scanning : scans whatever page is put below the scanner
- Sound on scan : just to acknowledge the user that a sheet was scanned (because when you autoscan you don't / can't really see your screen)
- Bonus : 
  - handling multiple PDF at once before uploading them. Like, maybe a list / grid where you can choose in what PDF the upcoming pages are going to go.
  - I forgot, I had an idea.
     - edit : it was probably about double page scanning, for copybooks

*Since I'm thinking about it, it's basically the features that I'm using the most on vFlat (which is a document scanner on the play store)

**Comment by SimSon2710** [comment]
As I think the user experience of 'Genius Scan' is pretty nice, at least regarding the scan process itself. Best feature is perhaps the 'focus-bubble' which is shown when one cuts/sizes the scanned document. Also good ist the format choice between color mode, b/w mode, foto mode and none. Last but not least, one can choose between different paper sizes (A4, A5...). 

Cheers and thanks for the beautiful app!

**Comment by bats77** [comment]
Indeed Genius scan is very fast, with a great autofocus and automatic contrast to enhance the quality ; I approve all what's written before.
I don't use the OCR over mobile, as I upload directly my files to server, which will treat them more efficiently.
I'm just a bit frustated about one point, for which I don't read nothing in discussions... In scan method instead of one round button, we shall have two concurrent buttons, one to pursue a multipage document, another one to batch scans aka many documents scanned in serial way (with automatic names)... to be treated later.
Multipage scanned documents shall have also an option to separate some page in (new) document.
I'm still trying many scanning apps, but none have this function fluently.

**Comment by astubenbord** [comment]
Hi, feel free to check out the latest pre-release with a new scanner: https://github.com/astubenbord/paperless-mobile/discussions/645 and let me know your thoughts in the linked discussion. The scanner is fully customizable and therefore suggestions and ideas for an improved workflow can be implemented more easily.
